sdcm/tester.py

```diff
@@ -546,6 +550,17 @@ def argus_collect_logs(self, log_links: dict[str, list[str] | str]):
         except Exception:  # pylint: disable=broad-except
             self.log.error("Error saving logs to Argus", exc_info=True)
 
+    def _init_data_validation(self):
```

Comment:
Why not put this data validation to external file/class (like you extracted `PartitionsValidationAttributes`) - you could encapsulate all the related functions to one place and not pollute `Tester` class which is complex enough.
So, tester class would have only reference to `DataValidator` (or whatever name you choose) and use it. Assuming we have only one `Tester` class instance, then this `DataValidator` could be an singleton for easier use.

If you'd gave also a good API and docstrings then it would be much easier to read, understand, fix, extend for future generations :)

Reply:
+1

Reply:
fixed. tester.py now only has one init line of: `self.partitions_attrs: PartitionsValidationAttributes | None = self._init_data_validation()`

Reply:
Actually not, I meant to extract all the partition data validation logic to separate module and all related things to it, so we don't expand `Tester` class anymore.
Still I see `validate_rows_per_partitions` and `collect_partitions_info` methods in `Tester` class.

Why not to create a class like `PartitionDataValidator` that would self-configure according to test settings and contain all related logic so it's easier to follow, test and use?

Reply:
ok, thanks @soyacz ,
moved validate_rows_per_partitions and collect_partitions_info methods from tester to `PartitionsValidationAttributes` in `table_metadata.py` now